### PR TITLE
Group 4 changes

### PR DESCRIPTION
**Pull request by Blitcut**
We've primarily reformatted the algorithms, added some new examples and two new exercises.

### CODE REVIEW

File: lecture6.tex [hunk 1]
```diff
@@ -221,7 +221,6 @@ \section{Graph distances}
 \section{Exercises}
 %TODO
```
Comment (vagdur) [inline lecture6.tex]: Should probably remove this comment as well.

As a general best practices thing, keeping to-do lists or notes to self in the actual code is going to come back to bite you. (I speak from experience here - my first paper is full of such notes and it is a *horrible unreadable mess* if you look at the TeX source, and the todos get forgotten or lost, not done. I have *regrets* about how that paper was written... 🥲) 

Try to use a separate system for that - GitHub has a built-in issue tracker that'd probably work pretty okay for a project like this.

Comment (Blitcut) [reply lecture6.tex]: Thanks for the advice! Should be fixed now.

**Comment by vagdur** [comment]
[lecture6_comments.pdf](https://github.com/vagdur/Grafteori-1MA170/files/13502122/lecture6_comments.pdf)
[comments_xca.pdf](https://github.com/vagdur/Grafteori-1MA170/files/13502124/comments_xca.pdf)

**Comment by Blitcut** [comment]
We've now implemented the feedback and added the solutions.

**Comment by vagdur** [comment]
You included one figure in your solutions to the exercises, but it seems you forgot to upload that figure here, so the TeX doesn't compile. Could you fix this?

**Comment by Blitcut** [comment]
Should be fixed.

**Comment by vagdur** [comment]
Thank you - as you can see I did a bunch of mostly very minor tweaks (the only "significant" change is probably to how paragraphs and spacing is handled in the exercise solutions) and merged your work into the main branch. I'll add a grade in Studium for you now.

**Pull request merged**